WordPress/pattern-directory
# Pattern Validation: Try checking block count, flagged title words

See #449 — Adds some automated checks based on the comments in this issue.

1. Check if there are more than 3 non-empty blocks
2. Check if there are less than 75 non-empty blocks
3. Check if the title contains any flagged words, like "test" or "my pattern"

This uses the same validation format as the other content validation, where submitting a pattern is totally blocked.

I also updated the tests a bit to clean up some of the repeated pattern-saving code (TIL about phpunit's `dataProvider` feature).

### Screenshots

When invalid content is submitted, it shows up in the submission modal and as a banner above the title, so it's clear how to fix the issue even when the modal is closed.

![error-notice](https://user-images.githubusercontent.com/541093/163271929-7de0424e-caed-47de-8219-243f41ec82d1.png)

### How to test the changes in this Pull Request:

1. Create or edit a pattern
2. Use some combination of invalid content
3. You should see the error both in the modal and above the title
4. Fix the issue & resubmit, it should work
